File: client/src/service/PhotoService.js
import axios from "axios";

export default {
    getAllPhotos() {
        return axios.get("/api/photos");
    },
    getPhotosByCategory(category) {
        return axios.get(`/api/photos/category/${category}`);
    },
    getPhotoById(id) {
        return axios.get(`/api/photos/${id}`);
    },
    createPhoto(photo) {
        return axios.post("/api/photos", photo);
    },
    deletePhoto(id) {
        return axios.delete(`/api/photos/${id}`);
    },

}

File: server/src/main/java/com/usforever/service/PhotoService.java
package com.usforever.service;

import com.usforever.model.Photo;

import java.util.List;

public interface PhotoService {
    List<Photo> getAllPhotos();
    List<Photo> getAllPhotosByCategory(String category);
    Photo getPhotoById(int id);
    Photo createPhoto(Photo newPhoto);
    int deletePhoto(int id);
}
